tools/AzDev/src/Services/CsprojReader.cs

```diff
@@ -0,0 +1,67 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AzDev.Services
+{
+    internal static class CsprojReader
+    {
+        private static IDictionary<string, Csproj> _cache = new Dictionary<string, Csproj>();
+
+        public static Csproj Parse(string content)
+        {
+            if (_cache.TryGetValue(content, out var result))
+            {
+                return result;
+            }
+
+            var lines = content.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+            var packageReferences = new List<string>();
+            var projectReferences = new List<string>();
+
+            foreach (var line in lines)
```

Comment:
any plan recursively getting the entire dependency tree? for example dependencies from common.netcore.dependencies.targets

Reply:
It's fairly complicated to fully parse the dependency tree.
Let's do it when necessary